CodeU_Assignement_1/src/codeu_assignement_1/main.java

```diff
@@ -0,0 +1,278 @@
+        SinglyLinkedList<Integer> my_list;
+        my_list = new SinglyLinkedList<>();
+        my_list.addFirst(1);
+        my_list.addFirst(2);
+        my_list.addFirst(3);
+        my_list.addFirst(4);
+        my_list.addFirst(4);
+        my_list.addFirst(5);
+        my_list.removeAt(5);
+        
+        System.out.println("> Last item is: " + my_list.get(my_list.size()-1));
+        System.out.println("> First item is: " + my_list.get(0));
+        System.out.println("> 2-3 items are: " + my_list.get(1) + ", " + my_list.get(2));
+        System.out.println("> Last was " + my_list.removeLast() + 
+                " and now is " + my_list.get(my_list.size()-1));
+        
+        if (my_list.size !=0)
+            System.out.println("> First was " + my_list.removeFirst() + 
+                " and now is " + my_list.get(0));
+        
+        if (my_list.size !=0)
+            System.out.println("> Last was " + my_list.removeLast() + 
+                " and now is " + my_list.get(my_list.size()-1));
+        
+        if (my_list.size !=0)
+            System.out.println("> First was " + my_list.removeFirst() + 
+                " and now is " + my_list.get(0));  
+        
+        if (my_list.size !=0)
+            System.out.println("> Last was " + my_list.removeLast() + 
+                " and now is " + my_list.get(my_list.size()-1));
+        
+        if (my_list.size !=0)
+            System.out.println("> First was " + my_list.removeFirst() + 
+                    " and now is " + my_list.get(0));
+    }
+        
+    static void run_test_Q2(String str_list, int k2last)
+    {
+        String[] intput_list = str_list.split(",");
+        
+        SinglyLinkedList<String> my_list;
+        my_list = new SinglyLinkedList<>(Arrays.asList(intput_list));
+        
+        LinkedList<String> _list;
+        _list = new LinkedList<>(Arrays.asList(intput_list));
+        
+        // Q2: Run and result
+        String res = q2_getKthElement(my_list, k2last); // Incase of error will throw exception before print
+        String _res = test_getKthElement(_list, k2last);
+        System.out.println("Q2: the " + k2last + "-th to the last element is " 
+                + res + ((res.compareTo(_res) == 0) ? " and correct result" : " and not correct result"));
+        
+    }
+
+    /*
+     * Works on Java LinkedList - to test the result is the 
+     * same as with my own container!
+     * Index starts from 0 to size-1
+     */    
+    static <E> E test_getKthElement(LinkedList<E> l, int k2last) 
+    {
+        if (l.isEmpty())
+            throw new IndexOutOfBoundsException(); // Any index is out of bound
+        
+        // Gets the k-th location out-of the k-th to the last location
+        int k = l.size() - k2last - 1;
+        if (k < 0 || k > l.size()) 
+            throw new IndexOutOfBoundsException();
+    
+        return l.get(k);
+    } // End Q2
+}

```

Comment:
The tests are generally a bit messy and verbose. I recommend playing around with a unit testing framework like JUnit to understand the modern preferred testing style.